### PR TITLE
Update evaluation logging test

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Updates a logging test to reduce the diff with https://github.com/Lightning-AI/lightning/pull/18895

cc @borda

### CODE REVIEW

File: tests/tests_pytorch/trainer/logging_/test_eval_loop_logging.py [hunk 1]
```diff
@@ -527,57 +527,47 @@ def test_step(self, batch, batch_idx):
...
-    assert set(get_metrics_at_idx(4)) == expected
-    assert set(get_metrics_at_idx(5)) == expected
-
-    assert get_metrics_at_idx(4)["valid_loss_0_step"] == model.val_losses[4]
-    assert get_metrics_at_idx(5)["valid_loss_0_step"] == model.val_losses[5]
-
-    assert set(get_metrics_at_idx(6)) == {"valid_loss_0_epoch", "valid_loss_1", "epoch"}
-
-    assert get_metrics_at_idx(6)["valid_loss_1"] == torch.stack(model.val_losses[4:]).mean()
+    assert get_metrics_at_idx(2)["valid_loss_0_step"] == model.val_losses[2]
```
Comment (awaelchli) [inline tests/tests_pytorch/trainer/logging_/test_eval_loop_logging.py:562]: maybe we should also assert get_metrics_at_idx(0) and (1)?

**Review by awaelchli — APPROVED** [review]

File: tests/tests_pytorch/trainer/logging_/test_eval_loop_logging.py (hunk 1 above)
Comment (carmocca) [reply tests/tests_pytorch/trainer/logging_/test_eval_loop_logging.py:562]: This is in the evaluation tests. Index 0 is the hparams and index 1 id for training metric. They are not relevant to the test

File: tests/tests_pytorch/trainer/logging_/test_eval_loop_logging.py [hunk 2]
```diff
@@ -527,57 +527,47 @@ def test_step(self, batch, batch_idx):
     trainer = Trainer(
         default_root_dir=tmpdir,
         logger=TensorBoardLogger(tmpdir),
-        limit_train_batches=2,
+        limit_train_batches=1,
```
Comment (Borda) [inline tests/tests_pytorch/trainer/logging_/test_eval_loop_logging.py:530]: I would keep at least two so we test iteration

Comment (carmocca) [reply tests/tests_pytorch/trainer/logging_/test_eval_loop_logging.py:530]: This is an evaluation logging tests. There's a thousand other tests that test more than 1 batch :joy:

**Review by Borda — APPROVED** [review]
